ui/input/src/moveCtrl.ts

```diff
@@ -48,7 +48,7 @@ export function makeMoveCtrl(root: RootCtrl, step: { fen: string }): MoveCtrl {
       initFen = step.fen;
       handlers.forEach(h => h(step.fen, cgState.movable.dests, yourMove));
     },
-    addHandler(h: MoveHandler | undefined) {
+    addHandler(h: MoveHandler) {
       if (!h) return;
```

Comment:
There's already a check here (`if (!h) return;`). If the `undefined` type isn't necessary, neither is that check.